integration-tests/quartz/src/test/java/io/quarkus/it/quartz/QuartzTestCase.java

```diff
@@ -20,6 +20,15 @@ public void testCount() throws InterruptedException {
         assertCounter("/scheduler/count/fix-8555");
     }
 
+    @Test
+    public void testJobData() throws InterruptedException {
+        // Wait at least 2 seconds for at least one run of the manual schedul and for job data to be persisted
+        Thread.sleep(2000);
```

Comment:
I'm not a fan of the `Thread.sleep()` usage in tests. Could we try to replace it with a `CountDownLatch`? The `QuartzTestCase` could inject the `ManualScheduledCounter`...

Reply:
I am not sure if injecting `ManualScheduledCounter` is going to work due to https://quarkus.io/guides/getting-started-testing#native-executable-testing as we've the `QuartzITCase` which is testing the native case. 

I could look to transform the `assertCounter` method to use `awaitility` to wait up to a certain counter or timeout in `x` seconds. I'll do it in a separate commit, in a couple of hours.